### DISCUSSION TITLE
Any way to use command cooldowns for slash commands?

### DISCUSSION THREAD

**Discussion by brotherelric** [discord-py-ui/discord-ui#105] [Q&A] [answered]
I would want a command I have in a slash cog to not be replying every time the user wants to, I would want a 30 second cooldown so the bot doesn't have to run many commands at once from various users.

**Comment by kvsxxx** [comment]
Cooldown support for slashcommands is currently limited to the  [**preview version**](https://github.com/discord-py-ui/discord-ui/tree/preview)

```py
from discord.ext import commands

# the `1` will indicate how often the command can be used before a cooldown is "activated"
# `30` is the amount of seconds for cooldown 
@slash_command(...)
@commands.cooldown(1, 30, commands.BucketType.user)
async def my_command(self, ctx):
   ...
```

**Reply by brotherelric** [reply]
This doesn't seem to work for me: no errors, command works perfectly fine..

**Comment by kvsxxx** [comment]
I'm really sorry for that, seems that cooldowns can't work on slashcommands because of a missing attribute (interactions don't have a `created_at` attribute).
I'll try to add something for cooldown on slashcommads

**Reply by brotherelric** [reply]
Sure! I'll be waiting.

**Comment by kvsxxx** [comment, marked as answer]
I added cooldown support to the [preview version](https://github.com/discord-py-ui/discord-ui/tree/preview) so if you want to use cooldowns you will either have to switch to the preview version or wait until it gets released

I've updated my original answer

Sorry for the inconvenience

**Reply by brotherelric** [reply]
Downloaded the preview. Few things to note:
-It tells me @slash_cog is deprecated, @slash_command worked fine so no problems
-I cannot send embeds. "discord.errors.HTTPException: 400 Bad Request (error code: 50006): Cannot send an empty message"
-Cooldowns with @commands.cooldown() doesn't work.

**Reply by kvsxxx** [reply]
You have to place the cooldown decorator before the slash_command decorator

I'll fix the embed issue, thank you for reporting that

**Reply by brotherelric** [reply]
Oh, that worked! Thank you!
Is there a way to send an specific message when a specific command is on cooldown? Different message when on-cooldown depending on the command.

**Reply by kvsxxx** [reply]
yes, I added something for this, you can do this by

```py
class MyCog(commands.Cog):
    @slash_command(...)
    @commands.cooldown(...)
    async def my_command(self, ctx):
        ...
    @my_command.cooldown
    async def my_cooldown_handler(self, ctx, cooldown_exception):
        await ctx.send(cooldown_exception) # just an example
```

`cooldown_exception` is of type `discord.ext.commands.errors.CommandOnCooldown`
